staging/src/github.com/kubeedge/beehive/pkg/core/core.go

```diff
@@ -84,3 +90,46 @@ func moduleKeeper(name string, moduleInfo *ModuleInfo, m common.ModuleInfo) {
 		beehiveContext.AddModuleGroup(name, moduleInfo.module.Group())
 	}
 }
+
+// localModuleKeeper starts and tries to keep module running when module exited.
+// Call EnableModuleRestart() to enable auto-restarting feature in alpha version.
+func localModuleKeeper(m *ModuleInfo) {
+	if !moduleRestartEnabled {
+		m.module.Start()
+		return
+	}
+
+	ctx := beehiveContext.GetContext()
+	backoffDuration := time.Second
+
+	// do if module exits
+	afterFunc := func() {
+		if r := recover(); r != nil {
+			klog.Errorf("module %s panicking: %v", m.module.Name(), r)
+		}
+		m.active = false
+		klog.Errorf("module %s exited, will restart in %ds", m.module.Name(), int(backoffDuration.Seconds()))
+	}
+
+	for {
+		func() {
+			defer afterFunc()
+			m.active = true
+			m.module.Start()
+		}()
+
+		select {
+		case <-ctx.Done():
+			klog.Infof("module %s shutdown", m.module.Name())
+			return
+		case <-time.After(backoffDuration):
+		}
+
+		if backoffDuration < 30*time.Second {
+			backoffDuration *= 2
+			if backoffDuration > 30*time.Second {
+				backoffDuration = 30 * time.Second
```

Comment:
We need to set the Max retry time, so process can exit after that, and the `backoffDuration` also should be configurable for users.

Reply:
We can make it configurable in the next phase of development. 

Now it's a feature in development. So we haven't add configurations yet.